src/dev/pca9702.rs

```diff
@@ -0,0 +1,194 @@
+//! Support for the `PCA9702` "8-Bit Input-Only Expander with SPI"
+//!
+//! Datasheet: https://www.nxp.com/docs/en/data-sheet/PCA9701_PCA9702.pdf
+//!
+//! The PCA9702 offers eight input pins, with an interrupt output that can be asserted when
+//! one or more of the inputs change state (enabled via the `INT_EN` pin). The device reads
+//! its inputs on each falling edge of `CS` and then presents them on `SDOUT` bit-by-bit as
+//! the clock (SCLK) rises.
+//!
+//! Because the PCA9702 is strictly input-only, there is no way to drive output values or
+//! configure directions. Consequently, calling methods that attempt to write or read back
+//! “set” states (e.g., `set()`, `is_set()`) will return an error.
+
+use crate::{PortDriver, SpiBus};
+use embedded_hal::spi::Operation;
+
+/// Error type for PCA9702 driver.
+/// We wrap both bus errors and "unsupported operation" errors here.
+#[derive(Debug)]
+pub enum Pca9702Error<E> {
+    /// Underlying SPI bus error
+    Bus(E),
+    /// Called an output-only function on an input-only device
+    OutputNotSupported,
+}
```

Comment:
By panicking on output, this error type becomes superfluous, so you can instead just use the bus error as the error type, making the code a bit simpler :)